src/NexusMods.App/Startup.cs

```diff
@@ -41,14 +66,45 @@ public static void Main(IServiceProvider provider, string[] args)
     }
 #pragma warning restore CS0028
 
+    private static void AppMain(Application app, string[] args)
+    {
+        _logger.LogInformation("Starting UI application");
+        ShowMainWindow();
+
+        // Start the app, linking to the global shutdown token
+        _logger.LogInformation("Starting application loop.");
+        app.Run(MainThreadData.GlobalShutdownToken);
+        _logger.LogInformation("Application loop ended.");
```

Comment:
Log event messages should be fragments, not sentences. Avoid a trailing period/full stop.